### PR TITLE
Primarybutton

### PR DESCRIPTION
**Pull request by maxime-carabina**
image

### CODE REVIEW

File: presets/massa-station-preset.js [hunk 1]
```diff
@@ -81,7 +97,7 @@ module.exports = {
           fontSize: "16px",
           fontWeight: "600",
           fontFamily: theme("fontFamily.Urbane"),
-          lineHeight: "19px",
+          // lineHeight: "19px",
```
Comment (pivilartisant) [inline presets/massa-station-preset.js:100]: why did you remove line height ?

**Review by pivilartisant — APPROVED** [review]
LTGM

File: presets/massa-station-preset.js (hunk 1 above)
Comment (maxime-carabina) [reply presets/massa-station-preset.js:100]: Sorry ! I change this.

File: presets/massa-station-preset.js [hunk 2]
```diff
@@ -39,6 +39,14 @@ module.exports = {
...
         info: colorLightBlue,
         success: colorBrand,
+        default_primary: colorDarkBlue,
+        hover_primary: colorBasicBlue,
+        pressed_primary: colorLightBlue,
+        disabled_primary: colorDarkGrey,
+        default_secondary: colorWhite,
+        hover_secondary: colorDarkBlue,
+        pressed_secondary: colorDarkBlue,
+        disabled_secondary: colorLightGrey,
```
Comment (mazmassa) [inline presets/massa-station-preset.js:49]: Why do we need that!? Colors are already there!? 🤔

File: src/components/Button/Button.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+const args = {
+  onClick: () => {
+    console.log("clicked");
+  },
+  iconStart: <FiArrowUpRight />,
+  iconEnd: <FiArrowUpRight />,
+};
+
+export const _PrimaryDefault = {
+  render: () => <Button {...args}>Button 1</Button>,
```
Comment (mazmassa) [inline src/components/Button/Button.stories.tsx:14]: Use the name of the button instead here: Primary, Secondary, etc...


```suggestion
  render: () => <Button {...args}>Primary</Button>,
```

File: presets/massa-station-preset.js (hunk 2 above)
Comment (maxime-carabina) [reply presets/massa-station-preset.js:49]: image

Lucas added all states in theme.
As the colours between primary and secondary are not the same for the states, I put them in the theme.
Do you have any suggestion to improve this?

File: src/components/Button/Button.stories.tsx (hunk 3 above)
Comment (maxime-carabina) [reply src/components/Button/Button.stories.tsx:14]: Indeed! It is more logical.

File: presets/massa-station-preset.js (hunk 2 above)
Comment (maxime-carabina) [reply presets/massa-station-preset.js:49]: image

I suggest, we can add it in addComponents section (look on screenshot) like this :
`
".btn-primary-default": {
          background: colorWhite,
        },
`

Same for all states.

Comment (mazmassa) [reply presets/massa-station-preset.js:49]: Ah got it!

I would suggest to ask Lucas about the color names and add that on the colors presset. Please, keep @Thykof aware of that too. (avoid _ on names too - please camelCase here in JS).

Comment (maxime-carabina) [reply presets/massa-station-preset.js:49]: Okay!
So if I understand correctly, we're sticking with what I did. But I change the "default_primary" to "btnPrimaryDefault"?

Or

I add ".btn-primary-default" (for example) in addComponent function to use it in the button component ?

Comment (mazmassa) [reply presets/massa-station-preset.js:49]: No, not exactlu btnPrimaryDefault. We can't have colors names that rely on components or action names. It needs to be more generic. Please, sync with Paul about that. I just explain that to him.

Comment (maxime-carabina) [reply presets/massa-station-preset.js:49]: Ok ! I will ask Paul to explain.

Thank you for your time ;)

File: src/components/Button/Button.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { Button } from "./Button";
+import { FiArrowUpRight } from "react-icons/fi";
+export default { title: "Components/Button", component: Button };
+
+const args = {
+  onClick: () => {
+    console.log("clicked");
+  },
+  iconStart: <FiArrowUpRight />,
+  iconEnd: <FiArrowUpRight />,
```
Comment (mazmassa) [inline src/components/Button/Button.stories.tsx:10]:
```suggestion
  preIcon: <FiArrowUpRight />,
  posIcon: <FiArrowUpRight />,
```

File: presets/massa-station-preset.js [hunk 5]
```diff
@@ -81,7 +85,7 @@ module.exports = {
           fontSize: "16px",
           fontWeight: "600",
           fontFamily: theme("fontFamily.Urbane"),
-          lineHeight: "19px",
+          // lineHeight: "19px",
```
Comment (mazmassa) [inline presets/massa-station-preset.js:88]: Is that intended!?

I just did the same review in Paul's MR. There's no reason to have a commented code in the codebase. If is not used, remove it.

File: src/components/Button/Button.test.tsx [hunk 6]
```diff
@@ -17,9 +17,9 @@ describe("Components | Buttons | PrimaryButton", () => {
       return <p data-testid="nested">something in nested</p>;
     };
     render(
-      <PrimaryButton>
+      <Button>
         <SomethingComponent />
```
Comment (mazmassa) [inline src/components/Button/Button.test.tsx:21]: `<SomeComponent />`

File: src/components/Button/Button.tsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  iconEnd?: JSX.Element;
+}
+
+export function Button(props: ButtonProps) {
+  const hoverClass = "hover:bg-tertiaryAlt";
+  const activeClass = "active:bg-secondaryAlt";
+  const disabledClass =
+    "disabled:bg-[#DADADA] disabled:cursor-not-allowed disabled:text-white";
+  const paddingClass = "px-4 py-3";
+  const sizeClass = "w-full h-[48px]";
```
Comment (mazmassa) [inline src/components/Button/Button.tsx:15]: Let's avoid custom values like: h-[48px]. You can replace by: `h-12`.  BTW if you remove the w-full isn't take the whole place by default!?

File: src/components/Button/Button.tsx [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  children?: React.ReactNode;
+  iconStart?: JSX.Element;
+  iconEnd?: JSX.Element;
+}
+
+export function Button(props: ButtonProps) {
+  const hoverClass = "hover:bg-tertiaryAlt";
+  const activeClass = "active:bg-secondaryAlt";
+  const disabledClass =
+    "disabled:bg-[#DADADA] disabled:cursor-not-allowed disabled:text-white";
```
Comment (mazmassa) [inline src/components/Button/Button.tsx:13]: Why do we need a custom color here!? We have all colors mapped in the codebase already. That's the aim of use Tailwind. Have you try that!?

File: src/components/Button/Button.tsx [hunk 9]
```diff
@@ -0,0 +1,35 @@
+import React, { ComponentPropsWithoutRef } from "react";
+
+export interface ButtonProps extends ComponentPropsWithoutRef<"button"> {
+  children?: React.ReactNode;
+  iconStart?: JSX.Element;
+  iconEnd?: JSX.Element;
```
Comment (mazmassa) [inline src/components/Button/Button.tsx:6]:
```suggestion
  preIcon?: JSX.Element;
  posIcon?: JSX.Element;
```

**Review by mazmassa — COMMENTED** [review]
Could you, please, provide a description and an image of the component on the description MR header?1 Like that I can have more context of this MR.

File: src/components/Button/Button.tsx (hunk 8 above)
Comment (maxime-carabina) [reply src/components/Button/Button.tsx:13]: image
image

I created a class name "infoAlt" and I used it in class

Comment (maxime-carabina) [reply src/components/Button/Button.tsx:13]: If it's okay for you, I push again on my branch

File: presets/massa-station-preset.js [hunk 10]
```diff
@@ -33,21 +30,63 @@ module.exports = {
     createThemes({
       light: {
         primary: colorWhite,
+        primaryAlt: colorDarkBlue,
         secondary: colorLightGrey,
+        secondaryAlt: colorBasicBlue,
         tertiary: colorDarkGrey,
-        brand: colorBrand,
+        brand: colorGreen,
```
Comment (mazmassa) [inline presets/massa-station-preset.js:37]: Same here...

File: presets/massa-station-preset.js [hunk 11]
```diff
@@ -81,7 +120,6 @@ module.exports = {
           fontSize: "16px",
           fontWeight: "600",
           fontFamily: theme("fontFamily.Urbane"),
-          lineHeight: "19px",
```
Comment (mazmassa) [inline presets/massa-station-preset.js:84]: This is out of date too!

File: src/components/Button/Button.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+  preIcon: <FiArrowUpRight />,
+  posIcon: <FiArrowUpRight />,
+};
+
+export const _Primary = {
+  render: () => (
+    <>
+      <Button {...primaryArgs}>Primary</Button>
+
+      <Button style={{ marginTop: 50 }} {...primaryArgs} disabled>
```
Comment (mazmassa) [inline src/components/Button/Button.stories.tsx:18]: Is there any reason that we are setting styles here!? 🤔

File: presets/massa-station-preset.js (hunk 10 above)
Comment (maxime-carabina) [reply presets/massa-station-preset.js:37]: Done too

File: src/global.css [hunk 13]
```diff
@@ -1,5 +1,8 @@
-@tailwind base;
-@tailwind components;
+@layer tailwind-base, mas-custom-styles;
+@layer tailwind-base {
+  @tailwind base;
+}
+
```
Comment (mazmassa) [inline src/global.css:5]: I have the feeling you are not rebasing. This was already done!

File: src/components/Button/Button.tsx [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+
+export function Button(props: ButtonProps) {
+  const paddingClass = "px-4 py-3";
+  const sizeClass = "w-full h-12";
+  const backgroundClass = "bg-c-default";
+  const textClass = "text-primary";
+  const hoverClass = "hover:bg-c-hover";
+  const activeClass = "active:bg-c-pressed";
+  const disabledClass =
+    "disabled:bg-c-disabled-1 disabled:text-primary disabled:cursor-not-allowed";
```
[the remark this answers is not in the document]
Comment (maxime-carabina) [reply src/components/Button/Button.tsx:17]: I think I will refactor this part when primary and secondary buttons is merged on main.
After that, I will create another issue to resolve this problem

**Comment by SlnPons** [comment]
@maxime-carabina the buttons don't match the designs. Is it normal? 
image

image

Maybe I don't know how to use the story book correctly. Just let me know.
cc @mazmassa

**Comment by mazmassa** [comment]
This is a closed MR... no idea why we are relying on that!
